docs/sources/flow/reference/components/otelcol.processor.memory_limiter.md

```diff
@@ -0,0 +1,107 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/otelcol.processor.memory_limiter
+title: otelcol.processor.memory_limiter
+---
+
+# otelcol.processor.memory_limiter
+
+`otelcol.processor.memory_limiter` is used to prevent out of memory situations
+on a telemetry pipeline. It performs periodic checks of memory usage, and will
+begin dropping data and forcing GC to reduce memory consumption if usage
+exceeds the defined limits.
+
+The `memory_limiter` component uses both soft and hard limits, where the latter
+is always equal or larger than the former. When memory usage goes above the
+soft limit, the processor component will start dropping data and returning
+errors to the preceding components in the pipeline. When usage exceeds the hard
+limit, the processor will force a garbage collection in order to try to free
+memory. When usage drops below the soft limit, the normal operation is resumed
+(data will not longer be dropped and no forced garbage collection will be
+performed).
+
+> **NOTE**: `otelcol.processor.memory_limiter` is a wrapper over the upstream
+> OpenTelemetry Collector `memorylimiter` processor. Bug reports or feature
+> requests will be redirected to the upstream repository, if necessary.
+
+Multiple `otelcol.processor.memory_limiter` components can be specified by
+giving them different labels.
+
+## Usage
+
+```river
+otelcol.processor.memory_limiter "LABEL" {
+  output {
+    check_interval = "1s"
+
```

Comment:
Here, we must either define `limit` _or_ `limit_percentage`. We explain this below, but do we want to signify this here, too?

Reply:
We should probably set one of them and then add comments saying the other can also be set.